librz/util/subprocess.c

```diff
@@ -132,29 +136,29 @@ static LPWCH override_env(const char *envvars[], const char *envvals[], size_t e
 	return ret;
 }
 
-static void remove_cr(char *str) {
-	char *start = str;
-	while (*str) {
-		if (str[0] == '\r' &&
-			!(str - start >= 4 && !strncmp(str - 4, RZ_CONS_CLEAR_SCREEN, 4))) {
-			memmove(str, str + 1, strlen(str + 1) + 1);
-			continue;
-		}
-		str++;
-	}
-}
-
 RZ_API RzSubprocess *rz_subprocess_start_opt(RzSubprocessOpt *opt) {
 	RzSubprocess *proc = NULL;
-	HANDLE stdin_read = NULL;
-	HANDLE stdout_write = NULL;
-	HANDLE stderr_write = NULL;
+	HANDLE stdin_read = GetStdHandle(STD_INPUT_HANDLE);
+	HANDLE stdout_write = GetStdHandle(STD_OUTPUT_HANDLE);
+	HANDLE stderr_write = GetStdHandle(STD_ERROR_HANDLE);
+	LPSTR lpFilePart;
+	char cmd_exe[MAX_PATH];
+
+	if (!rz_file_exists(opt->file) && NeedCurrentDirectoryForExePathA(opt->file)) {
+		DWORD len;
+		if ((len = SearchPath(NULL, opt->file, ".exe", sizeof(cmd_exe), cmd_exe, &lpFilePart)) < 1) {
+			RZ_LOG_DEBUG("SearchPath failed for %s\n", opt->file);
+			return NULL;
+		}
+	} else {
+		snprintf(cmd_exe, sizeof(cmd_exe), "%s", opt->file);
+	}
 
 	char **argv = calloc(opt->args_size + 1, sizeof(char *));
 	if (!argv) {
 		return NULL;
 	}
-	argv[0] = (char *)opt->file;
+	argv[0] = ""; // a space is required to work correctly.
```

Comment:
why the `cmd_exe` is not placed in `argv[0]`?

Reply:
because we are calling `CreateProcessA` properly. check MS guide.

Reply:
`""` is not a "space" though, it's an empty string.